Fix issues #5 and #13

cshaa opened the pull request:
Update html:meta, split html:link and html:a to html:alter, html:include, html:link and html:anch

Closes #13
Closes #5

cshaa committed:
Update html:meta, split html:link and html:a to html:alter, html:include, html:link and html:anch

Closes #13
Closes #5

cshaa committed:
Changing `html:include/@href` to `html:include/@src` - it's not a reference, it's path to the source.

OscarGodson commented:
@m93a without a target tho how would JavaScript know the target of the element? A common use case: http://www.ajaxf1.com/tutorial/ajax-file-upload-tutorial.html

OscarGodson commented on README.md:
```diff
@@ -148,23 +153,50 @@ _Example:_
+##### `<html:alter>`
 
-This links external documents and scripts such as CSS, JavaScript, RSS, favicons, etc. to the current document. Equivalent to the current `<link>` tag. This tag takes the following attributes:
+This links document's alternatives in other formats, eg. PDF, RTF, RSS or Atom feed. It's similar to the current HTML `link` tag.
```
Why would you have a different link just for documents? Why shouldn't the browser handle the displaying of these like it does now? What's considered an "alternative document" and how would you know what is and isn't one?

cshaa replied:
Current "link" was made to do that but little by little it became more of "include" than a real link. If you use <link> as a reference to RSS, the browser just shows _feed_ button and doesn't even download the file. But if you use it as a link to css stylesheet, browser downloads the file, includes it to the page and changes its content.

In layman's terms: `include` changes the document, `alter` doesn't

---

Or maybe we can use `meta` instead of `alter`?

OscarGodson replied:
So, basically this link just forces a download?

OscarGodson replied:
I'm assuming, yes? If so there's no need for an entirely different element for that. Just have a like `download` attribute that's already in HTML now.

cshaa replied:
include = make it a part of this document

OscarGodson commented on README.md:
````diff
@@ -148,23 +153,50 @@ _Example:_
+</html:html>
+```
+
+##### `<html:include>`
````
+1

cshaa committed:
Change image location to the current OscarGodson's HTML6.md

cshaa commented:
Now, my pull request should point to `_includes/HTML6.md`, not `README.md`

cshaa commented on _includes/HTML6.md:
```diff
@@ -187,15 +219,40 @@ _Example:_
-- `name`
-- `target` (can be `blank`, `parent`, `top` or `self`)
+- `href`: The link to the target document, can be relative or absolute.
+- `target`: Which window will open the document (can be `blank`, `parent`, `top` or `self`, default is `self`). **This attribute is obsolete**.
```
OscarGodson:

```
@m93a without a target tho how would JavaScript know the target of the element?
A common use case:
http://www.ajaxf1.com/tutorial/ajax-file-upload-tutorial.html
```

cshaa replied:
@OscarGodson 
I would make a `<button>` with onclick and use JavaScript XMLHttpRequest and File API.
A common use case:
http://www.matlus.com/html5-file-upload-with-progress/

OscarGodson replied:
If you only have to support new browsers, that'd be fine :)

cshaa replied:
Old browsers won't support HTML6 :)

philandy replied:
Oscar HTML6 doesn't have to worry about backwards compatibility because that would be up to the browsers.

cshaa commented:
@OscarGodson would you merge it to the master, please?

OscarGodson commented:
I'm still waiting on responses from you in the PR :) I'm not sold on your explanation of link and removing `a`.

cshaa commented:
Oh, that's a reaction to #5. It's about splitting `html:a` to `html:anch` and `html:link`.

OscarGodson commented on _includes/HTML6.md:
````diff
@@ -187,15 +219,40 @@ _Example:_
+</html:html>
+```
+
+##### `<html:anch>`
````
You could make this into it's own PR and ask people how they like it, but I don't see any benefit to this over just allowing `a`/`link`. In both cases it makes semantic sense. If we had a `link` element it'd mean `link to this page, and this section`. You'd also need links like `http://someothersite.com/#foo` to work anyway so making them totally different links seems confusing.

OscarGodson replied:
I think if you remove this and make it it's own PR and get feedback I'll accept the PR :)

cshaa replied:
Ok, rollback to html:a done. Adding "for" attribute to it so you can link elements using css selectors. It would be great to have a possibility to link somebody to the third `div.rc` of `google.com/#q=foo`.

OscarGodson commented on _includes/HTML6.md:
````diff
@@ -148,23 +153,50 @@ _Example:_
 ```
 
-##### `<html:link>`
+##### `<html:alter>`
````
I see. Makes sense to me now. I'll +1

cshaa committed:
Rollback to html:a (merge html:link and html:anch)

As OscarGodson said:
You could make this into it's own PR and ask people how they like it, but I don't see any benefit to this over just allowing a/link. In both cases it makes semantic sense. If we had a link element it'd mean link to this page, and this section. You'd also need links like http://someothersite.com/#foo to work anyway so making them totally different links seems confusing.

cshaa commented:
@OscarGodson is there anything you don't agree with? :)  
Actually, writing specifications began to be a good fun for me, so I'm ready to correct it.

cshaa committed:
Correcting a misspell.

risenomore commented:
"This begins an HTML's head. Equivelent to the current &lt;html&gt; tag."

Should read: "This begins an HTML's head. Equivelent to the current <b>&lt;head&gt;</b> tag."

cshaa committed:
fixed

@risenomore True story :)
